### DISCUSSION TITLE
Switching Quire versions or upgrading old projects?

### DISCUSSION THREAD

**Discussion by swroberts** [thegetty/quire#711] [General] [answered]
As we prepare to update our workflows to Quire 1.0, we're faced with the question of how to deal with older projects. We have many Quire projects using 0.20.4. Quire 1.0 will not build these older projects, so if they need updating, we need to use 0.20.4.

So: do we upgrade all the old projects to 1.0 or is there a way to install two versions of Quire and switch between them to update older projects?

Has anyone encountered this or have recommendations?

**Comment by geealbers** [comment, marked as answer]
@swroberts We run macOS and use Node Version Manager ([nvm](https://github.com/nvm-sh/nvm)). It works great for us. There's also a [nvm-windows](https://github.com/coreybutler/nvm-windows) which is a different tool, but should provide the same functionality. Using a version manager for Node let's you have different versions of Node on your computer, that you can easily switch back and forth from, one that you use for Quire 0.20.4 projects, and another for 1.0.0 projects. Here's our write up on using nvm specifically:

1. Uninstall your current version of Quire with `npm uninstall -g @thegetty/quire-cli`

2. Install nvm with the following command:

    ```
    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
    ```

3. Verify with this command, which should return `nvm`. If it does not, see the Troubleshooting info at https://github.com/nvm-sh/nvm#troubleshooting-on-macos.

    ```
    command -v nvm
    ```

4. Install the versions of node you want to use (14 for Quire 0.20.4 and earlier; 18 for Quire 1.0.0-rc.0 and later):

    ```
    nvm install 14.18.1
    ```

    ```
    nvm install 18.12.1
    ```
5. To choose/change a Node version to run use `nvm use 14` or `nvm use 18`. This will be the version used for as long as that Terminal window is open, or until you change it again. (Optionally, set a default version to use with `nvm alias default 14.18.1` or `nvm alias default 18.12.1`. This default will be the one used every time you open a new Terminal window.) 

6. Choose Node 14 with `nvm use 14` and then run `npm install -g @thegetty/quire-cli@0.20.4`

7. Choose Node 18 with `nvm use 18` and then run `npm install -g @thegetty/quire-cli` 

Thost last commands install the needed versions of Quire and associates them with those two versions of Node. You can then switch back and forth between Quire versions by switching back and forth between Node versions. So, `nvm use 14` and you're good to go with and 0.20.4 projects, and `nvm use 18` and you can work on 1.0.0 projects.

**Comment by swroberts** [comment]
Thank you so much, @geealbers! This is exactly what we needed to know. Solutions like this save a lot of work!
